lxl-web/src/lib/utils/cslFromMainEntity.ts

```diff
@@ -0,0 +1,118 @@
+	'publisher-place': `join(', ', publication[].place[].computedLabel)`,
+	issued: `{"date-parts": [[ (publication[0].year || '') ]] }`,
+	'number-of-pages': `join(', ', extent[].computedLabel)`,
+	edition: `editionStatement`,
+	language: `join(', ', instanceOf.language[].computedLabel)`,
+	DOI: `join(', ', identifiedBy[?"@type"=='DOI'].value)`,
+	ISBN: `join(', ', identifiedBy[?"@type"=='ISBN'].value)`,
+	ISSN: `join(', ', identifiedBy[?"@type"=='ISSN'].value)`,
+	URL: `join(', ', associatedMedia[*].uri[])`,
+	abstract: `join(', ', summary[].computedLabel)`,
+	keyword: `join(', ', instanceOf.subject[].computedLabel)`,
+	issue: `part`,
+	volume: `join(', ', [?"@type"=='Serial'].part[?"@type"=='Monograph'].mainEntity.hasTitle[].hasPart[].partNumber)`,
+
+	author: `instanceOf.contribution[?contains((role[]."@id" || [role."@id"]), 'https://id.kb.se/relator/author')].{family: agent.familyName, given: agent.givenName}`,
+	editor: `instanceOf.contribution[?contains((role[]."@id" || [role."@id"]), 'https://id.kb.se/relator/editor')].{family: agent.familyName, given: agent.givenName}`,
+	composer: `instanceOf.contribution[?contains((role[]."@id" || [role."@id"]), 'https://id.kb.se/relator/composer')].{family: agent.familyName, given: agent.givenName}`,
+	director: `instanceOf.contribution[?(contains((role[]."@id" || [role."@id"]),'https://id.kb.se/relator/filmDirector') || contains((role[]."@id" || [role."@id"]),'https://id.kb.se/relator/director') || contains((role[]."@id" || [role."@id"]),'https://id.kb.se/relator/televisionDirector'))].{family: agent.familyName, given: agent.givenName}`,
+	illustrator: `instanceOf.contribution[?contains((role[]."@id" || [role."@id"]), 'https://id.kb.se/relator/illustrator')].{family: agent.familyName, given: agent.givenName}`,
+	interviewer: `instanceOf.contribution[?contains((role[]."@id" || [role."@id"]), 'https://id.kb.se/relator/interviewer')].{family: agent.familyName, given: agent.givenName}`,
+	translator: `instanceOf.contribution[?contains((role[]."@id" || [role."@id"]), 'https://id.kb.se/relator/translator')].{family: agent.familyName, given: agent.givenName}`
+};
+
+export function cslFromMainEntity(mainEntity: FramedData, vocabUtil: VocabUtil): CSLJSON[] {
+	const result: Partial<CSLJSON> = {};
+
+	for (const [key, expr] of Object.entries(CSL_KBV_MAPPING) as [keyof CSLJSON, string][]) {
+		try {
+			const value = jmespath.search(mainEntity, expr);
+			if (value) {
+				result[key] = value;
+			}
+		} catch {
+			continue;
+		}
+	}
+
+	// Look for PrimaryContribution to put as author as fallback
+	if (!result?.author?.length) {
+		let PrimaryContribution: CSLName[] | undefined;
+		try {
+			PrimaryContribution = jmespath.search(
+				mainEntity,
+				`instanceOf.contribution[?"@type"=='PrimaryContribution'].{family: agent.familyName, given: agent.givenName}`
+			);
+		} catch {
+			// do nothing
+		}
+
+		if (PrimaryContribution) {
+			result.author = PrimaryContribution;
+		}
```

Comment:
I think we can fall back to any contribution after this

https://libris-qa.kb.se/katalogisering/5nnzsk2s3xhwbfbj

https://beta.libris-qa.kb.se/find?_q=contribution%3A*+NOT+contribution.%40type%3A%22PrimaryContribution%22+NOT+contribution.role%3A*+&_limit=20&_offset=0&_sort=&_spell=true